ppa/archive/templates/archive/snippets/cluster_work.html

```diff
@@ -1,5 +1,5 @@
 {# NOTE: expects cluster_id to be passed in #}
 <div class="cluster-work icon container" >
     {# TODO: include count if we can easily get it ; search and browse x digitized works within cluster #}
-    <a href="{% url "archive:list" %}?cluster={{ cluster_id }}">search and browse within cluster</a>
+    <a href="{% url "archive:list" %}?cluster={{ cluster_id }}{% if query %}&query={{ query }}{% endif %}">search and browse within cluster</a>
```

Comment:
I think we need to preserve all query parameters (keyword, author, title, date range, even collection).

If you look at the pagination template snippet, we're using a custom template tag called `querystring_replace` that lets us preserve all the search query args but replace the pagination. I think the same thing is relevant here.

Reply:
Hm. I was assuming if we're searching _within_ a cluster, the author, title, date range, and collection would also be the same for each text within the cluster. (Since from Mary's description significant differences in those things would constitute a different cluster). But I suppose we don't need to make that assumption in the code.